cuqi/experimental/mcmc/_rto.py

```diff
@@ -85,36 +81,38 @@ def data(self):
     @SamplerNew.target.setter
     def target(self, value):
         """ Set the target density. Runs validation of the target. """
-        # Accept tuple of inputs and construct posterior
-        if isinstance(value, tuple) and len(value) == 5:
-            # Structure (data, model, L_sqrtprec, P_mean, P_sqrtprec)
-            data = value[0]
-            model = value[1]
-            L_sqrtprec = value[2]
-            P_mean = value[3]
-            P_sqrtprec = value[4]
-
-            # If numpy matrix convert to CUQI model
-            if isinstance(model, np.ndarray) and len(model.shape) == 2:
-                model = cuqi.model.LinearModel(model)
-
-            # Check model input
-            if not isinstance(model, cuqi.model.LinearModel):
-                raise TypeError("Model needs to be cuqi.model.LinearModel or matrix")
-
-            # Likelihood
-            L = cuqi.distribution.Gaussian(model, sqrtprec=L_sqrtprec).to_likelihood(data)
-
-            # Prior TODO: allow multiple priors stacked
-            #if isinstance(P_mean, list) and isinstance(P_sqrtprec, list):
-            #    P = cuqi.distribution.JointGaussianSqrtPrec(P_mean, P_sqrtprec)
-            #else:
-            P = cuqi.distribution.Gaussian(P_mean, sqrtprec=P_sqrtprec)
-
-            # Construct posterior
-            value = cuqi.distribution.Posterior(L, P)
-        super(LinearRTONew, type(self)).target.fset(self, value)
-        self._precompute()
+        if value is not None:
+            # Accept tuple of inputs and construct posterior
+            if isinstance(value, tuple) and len(value) == 5:
+                # Structure (data, model, L_sqrtprec, P_mean, P_sqrtprec)
+                data = value[0]
+                model = value[1]
+                L_sqrtprec = value[2]
+                P_mean = value[3]
+                P_sqrtprec = value[4]
+
+                # If numpy matrix convert to CUQI model
+                if isinstance(model, np.ndarray) and len(model.shape) == 2:
+                    model = cuqi.model.LinearModel(model)
+
+                # Check model input
+                if not isinstance(model, cuqi.model.LinearModel):
+                    raise TypeError("Model needs to be cuqi.model.LinearModel or matrix")
```

Comment:
Just a small comment, these few lines (98-100) seem to better fit in `validate_target`

Reply:
I suppose so. although I can't see a nice way to do it. I would actually rather get rid of the tuple option for target similar to the other samplers. I got rid of it now. thoughts?

Reply:
I think this is a really good idea, to make target only posterior type